internal/venice-client-common/src/main/java/com/linkedin/venice/schema/SchemaUtils.java

```diff
@@ -100,4 +101,77 @@ public static GenericRecord createGenericRecord(Schema originalSchema) {
 
     return newRecord;
   }
+
+  /**
+   * Annotate all the top-level map field of the input schema to use Java String as key.
+   * @param schema the input value schema to be annotated.
+   * @return Annotated value schema.
+   */
+  public static Schema annotateStringMapInValueSchema(Schema schema) {
+    // Create duplicate schema here in order not to create any side effect during annotation.
+    Schema replicatedSchema = AvroSchemaParseUtils.parseSchemaFromJSONStrictValidation(schema.toString());
+    if (replicatedSchema.getType().equals(Schema.Type.RECORD)) {
+      for (Schema.Field field: replicatedSchema.getFields()) {
+        if (field.schema().getType().equals(Schema.Type.MAP)) {
+          annotateMapSchema(field.schema());
+        }
+      }
+    }
+    return replicatedSchema;
+  }
+
+  /**
+   * Annotate all the top-level map field of the partial update schema to use Java String as key. This method will make
+   * sure field update and collection merging operations of a top-level map field are annotated.
+   * @param schema the input partial update schema to be annotated.
+   * @return Annotated partial update schema.
+   */
+  public static Schema annotateStringMapInDerivedSchema(Schema schema) {
+    // Create duplicate schema here in order not to create any side effect during annotation.
+    Schema replicatedSchema = AvroSchemaParseUtils.parseSchemaFromJSONStrictValidation(schema.toString());
+    if (replicatedSchema.getType().equals(Schema.Type.RECORD)) {
+
+      for (Schema.Field field: replicatedSchema.getFields()) {
+        if (field.schema().isUnion()) {
+          for (Schema unionBranchSchema: field.schema().getTypes()) {
+            // Full update request for Map field.
+            if (unionBranchSchema.getType().equals(Schema.Type.MAP)) {
+              annotateMapSchema(unionBranchSchema);
+            } else if (unionBranchSchema.getType().equals(Schema.Type.RECORD)) {
+              for (Schema.Field updateOpField: unionBranchSchema.getFields()) {
+                if (updateOpField.schema().getType().equals(Schema.Type.MAP)) {
```

Comment:
Map operation contains a MapDiff, which is a list of `String`, shall we add annotation to that list as well?
Please double check whether DCR will use MapDiff as a String list or not.

Reply:
Good catch. I think I missed this one. Will add it.

Reply:
Good catch, previous version does have issue with this. I added patch to address that.